src/nacl/bindings/crypto_secretstream.py

```diff
@@ -73,13 +73,15 @@ class crypto_secretstream_xchacha20poly1305_state:
 
     def __init__(self) -> None:
         """Initialize a clean state object."""
-        self.statebuf = ffi.new(
+        # NOTE: the members below aren't `bytearray` objects, but cffi `cdata` objects
+        # which own an array of unsigned chars. Is there a better annotation?
```

Comment:
I considered `typing.ByteString` here, but 
- that's immutable, but one of the test writes to it
-  the docs [suggest](https://docs.python.org/3/library/typing.html?highlight=typing#typing.ByteString) that `bytes` in an annotation is really shorthand for a ByteString anyway.
    >  class typing.ByteString(Sequence[int])
    > 
    >   A generic version of collections.abc.ByteString.
    >
    >    This type represents the types bytes, bytearray, and memoryview of byte sequences.
    >
    >    As a shorthand for this type, bytes can be used to annotate arguments of any of the types mentioned above.

`bytearray` seemed like the least bad option, but would welcome input here!

Reply:
In cryptography we've chosen to leave these untyped for now with a hope of typing `ffi.new` itself in the future. I suppose if we do that then we'd catch these cases here, but I'm not hugely comfortable with picking a "closest approximation" type. e.g. you can't call `bytes()` on this buffer, which you could on a `bytearray`.

Reply:
That makes sense.

The only place (AFAICS) that the statebuf gets mutated by Python is [here](https://github.com/DMRobertson/pynacl/blob/6c4f3a2bcb18abc20bf294ed12d1b8f56bfdcf6f/tests/test_secretstream.py#L242-L245), and that's just to copy the data from another ffi `unsigned char[]`.

Perhaps it would be safer to use `Sequence[int]` here instead of `bytearray` and type-ignore the part of the test that writes to this buffer. (I stuck a breakpoint in at the snippet linked above, where `state_save[0]` was the int `68` as opposed to `b"\x68"`.)

Reply:
Actually, `ByteString` seems like a better fit: it inherits from `Sequence[int]`, but additionally expresses that the ints are all going to be byte-sized.

Reply:
I've had a go at this in bc42daf.